lndclient/router_client.go

```diff
@@ -226,6 +229,17 @@ func (r *routerClient) trackPayment(ctx context.Context,
 		for {
 			payment, err := stream.Recv()
 			if err != nil {
+				// If we get an EOF error, the payment has
+				// reached a final state and the server is
+				// finished sending us updates. We close both
+				// channels to signal that we are done sending
+				// values on them and return.
+				if err == io.EOF {
+					close(statusChan)
+					close(errorChan)
+					return
+				}
```

Comment:
Passing back your own question: are changes needed anywhere we use `trackPayment` because of this?

Reply:
Did a quick audit of our usage of  `TrackPayment` and `trackPayment` (called by `SendPayment`):
`lsat/interceptor.trackPayment`: exits on success/timeout, ok
`waitForHtlcSpendConfirmed`: this change, accounts for channel close
`payInvoiceAsync`: exits on success/timeout, ok

Also went through the other projects that use `lndclient` and they all follow the pattern of exiting once the payment is complete.